modules/webclient/src/main/scala/scaladex/client/ActiveNavObserver.scala

```diff
@@ -11,18 +13,34 @@ import scaladex.dom.IntersectionObserver
   * list items in nav.
   */
 object ActiveNavObserver {
-  private val observer: IntersectionObserver = IntersectionObserver { entry =>
-    val id = entry.target.getAttribute("id")
-    val navLink = document.querySelector(s"""nav li a[href="#$id"]""").asInstanceOf[Link]
-    val navLI = navLink.parentElement.asInstanceOf[LI]
+  def start(): Unit = {
+    val sectionsAndNavItem =
+      document
+        .querySelectorAll("section[id]")
+        .toSeq
+        .collect { case e: Element => e }
+        .flatMap { section =>
+          val id = section.getAttribute("id")
+          val link = document.querySelector(s"""nav li a[href="#$id"]""").asInstanceOf[Link]
+          Option(link).map(l => section -> l.parentElement)
+        }
 
-    if (entry.intersectionRatio > 0) navLI.classList.add("active")
-    else navLI.classList.remove("active")
+    var debounceUpdate: timers.SetTimeoutHandle = null
+    val observer = IntersectionObserver { _ =>
+      // ignore the observed entry and update all sections instead
+      debounceUpdate = timers.setTimeout(150) {
+        timers.clearTimeout(debounceUpdate)
```

Comment:
Hum, IIUC the `clearTimeout` must come before `debounceUpdate = setTimeout` (but inside the lambda passed to `IntersectionObserver`), shouldn't it?

Reply:
:man_facepalming: